app/components/post_list/date_separator/index.tsx

```diff
@@ -13,7 +13,7 @@ import {typography} from '@utils/typography';
 
 type DateSeparatorProps = {
     date: number | Date;
-    style?: StyleProp<ViewStyle>;
+    style?: StyleProp<TextStyle>;
```

Comment:
In order to make this type safe, you can use the Intersection utility type we have like this:

```
style?: StyleProp<Intersection<TextStyle, ViewStyle>>;
```

That way, we don't need the casting, and we make sure we don't pass styles that may create issues.